src/input.rs

```diff
@@ -27,9 +28,21 @@ impl InputContext {
             previous_key_state: FnvHashSet::default(),
             current_mouse_state: FnvHashSet::default(),
             previous_mouse_state: FnvHashSet::default(),
+            current_text_input: None,
             mouse_position: Vec2::zeros(),
         }
     }
+
+    pub(crate) fn cleanup_after_state_update(&mut self) {
+        self.previous_key_state = self.current_key_state.clone();
+        self.current_text_input = None;
+    }
+}
+
+/// Returns the text that the user entered this tick.
+/// This will match the user's keyboard and OS settings.
+pub fn get_text_input(ctx: &Context) -> &Option<String> {
+    &ctx.input.current_text_input
```

Comment:
`Option<&str>` would be a little nicer of a type signature - you could get that by doing `ctx.input.current_text_input.as_ref().map(String::as_str)`.

Reply:
I actually had that the first time, but figured that it was messier than returning an `&Option<String>`

Reply:
A little messier on our end, but hopefully a little clearer in the docs :)